strum_macros/src/helpers/type_props.rs

```diff
@@ -1,87 +1,56 @@
-use std::convert::From;
+use proc_macro2::TokenStream;
+use quote::quote;
 use std::default::Default;
-use syn::{DeriveInput, Lit, Meta, Path};
+use syn::{DeriveInput, Ident, Path};
 
 use crate::helpers::case_style::CaseStyle;
-use crate::helpers::has_metadata::HasMetadata;
-use crate::helpers::{MetaHelpers, NestedMetaHelpers};
+use crate::helpers::metadata::{DeriveInputExt, EnumDiscriminantsMeta, EnumMeta};
 
 pub trait HasTypeProperties {
     fn get_type_properties(&self) -> syn::Result<StrumTypeProperties>;
 }
 
-#[derive(Debug, Clone, Eq, PartialEq, Default)]
+#[derive(Debug, Clone, Default)]
 pub struct StrumTypeProperties {
     pub case_style: Option<CaseStyle>,
     pub discriminant_derives: Vec<Path>,
-    pub discriminant_name: Option<Path>,
-    pub discriminant_others: Vec<Meta>,
+    pub discriminant_name: Option<Ident>,
```

Comment:
I'm trying to think of any places where a path would be valid here, but I don't think it ever is so this is good.